Sources/MongoSwift/WriteConcern.swift

```diff
@@ -113,15 +105,14 @@ public class WriteConcern: Codable {
                 if wNumber < 0 {
                     throw UserError.invalidArgumentError(message: "Invalid value: w=\(w) cannot be negative.")
                 }
-                mongoc_write_concern_set_w(self._writeConcern, wNumber)
```

Comment:
so in this switch statement, all we really care about now is the case where it's `.number`, since we need to validate the stored number is >= 0. 

otherwise, all we end up doing in every case is setting `self.w` to `w`. we don't need to reconstruct the appropriate cases like we are now with `self.w = .tag(wTag)` and `self.w = .majority`.

I think we can simplify this logic and get rid of the switch statement using a `case let` (kinda funky but cool syntax to match a single case):
```swift
if case let .number(wNumber) = w {
    // error if wNumber < 0
} 

self.w = w
```

note this should also handle the case where the `w` parameter is nil - the `if` will evaluate to false and `self.w` will correctly be set to `nil`.

Reply:
Oh, yeah, I did find it kind of awkward having to use switch statement to extract the  `.number`, just didn't know you could use `case let` by itself. That's neat, thanks! 😄 

Updated.